artifactory/services/utils/artifactoryutils.go

```diff
@@ -242,47 +242,73 @@ func createBodyForLatestBuildRequest(buildName, buildNumber string) (body []byte
 }
 
 func filterAqlSearchResultsByBuild(specFile *ArtifactoryCommonParams, reader *content.ContentReader, flags CommonConf, itemsAlreadyContainProperties bool) (*content.ContentReader, error) {
-	var aqlSearchErr error
+	var artifactsAqlSearchErr, dependenciesAqlSearchErr error
 	var readerWithProps *content.ContentReader
-	var buildArtifactsSha1 map[string]int
+	buildArtifactsSha1 := make(map[string]int)
+	buildDependenciesSha1 := make(map[string]int)
 	var wg sync.WaitGroup
-	// If 'build-number' is missing in spec file, we fetch the laster from artifactory.
+	wg.Add(2)
+	// If 'build-number' is missing in spec file, we fetch the latest from artifactory.
 	buildName, buildNumber, err := getBuildNameAndNumberFromBuildIdentifier(specFile.Build, flags)
 	if err != nil {
 		return nil, err
 	}
 
-	wg.Add(1)
-	// Get Sha1 for artifacts by build name and number
 	go func() {
-		buildArtifactsSha1, aqlSearchErr = fetchBuildArtifactsSha1(buildName, buildNumber, flags)
-		wg.Done()
+		// Get Sha1 for artifacts.
+		defer wg.Done()
+		if !specFile.ExcludeArtifacts {
+			buildArtifactsSha1, artifactsAqlSearchErr = fetchBuildArtifactsOrDependenciesSha1(buildName, buildNumber, flags, true)
+		}
 	}()
 
-	if !itemsAlreadyContainProperties {
-		// Add properties to the previously found artifacts (in case properties haven't already fetched from Artifactory)
-		readerWithProps, err = searchProps(specFile.Aql.ItemsFind, "build.name", buildName, flags)
-		if err != nil {
-			return nil, err
-		}
-		defer readerWithProps.Close()
-		tempReader, err := loadMissingProperties(reader, readerWithProps)
-		if err != nil {
-			return nil, err
+	go func() {
+		// Get Sha1 for dependencies.
+		defer wg.Done()
+		if specFile.IncludeDeps {
+			buildDependenciesSha1, dependenciesAqlSearchErr = fetchBuildArtifactsOrDependenciesSha1(buildName, buildNumber, flags, false)
 		}
-		defer tempReader.Close()
+	}()
+
+	if specFile.ExcludeArtifacts || itemsAlreadyContainProperties {
```

Comment:
This if condition seems wrong, because if itemsAlreadyContainProperties is true, we shouldn't collect the properties.

Reply:
Right, the properties collection happens outside of the 'if' statement.
It reaches the properties collection only if we want to include artifacts, and the properties are missing from the search results.